js_hw_4/task_01.js

```diff
@@ -0,0 +1,30 @@
+// 1. Создайте функцию getEmployeeInfo()
+// 2. В функции создайте массив имен сотрудников 5шт (Имена придумать самим)
+// 3. В функции создайте массив зарплат сотрудников 5 шт(цифры придумать самим)
+// 4. Функция должна принимать 1 аргумент - имя сотрудника
+// 5. Функция должна возвращать новый массив, где первый элемент - имя сотрудника, второй - его зарплата
+// 6. Для поиска ответа функции нужно найти индекс сотрудника в массиве имен. Зарплату взять с ТЕМ ЖЕ индексом что и имя
+// 7. Для возврата из функции создайте массив, методом .push поместите в него имя и зарплату, и через return верните созданный массив
+// 8. Если такое имя сотрудника в массиве не найдется - вернуть null
+
+function getEmployeeInfo(name) {
+    if(name!== undefined && name!== null && name.length && !+name){
+        const names = ["Bill", "Alex", "Anna", "Jhonn", "Antony"];
+    const salaries = [1500, 3000, 1200, 1700, 850];
+    const index = names.indexOf(name);
```

Comment:
```suggestion
    const index = names.indexOf(name);
    if(index === -1) return null
```

Reply:
What's wrong with my implementation? return index === -1 ? null: resultArr

Reply:
If index is `-1` immediately return the `null`, no need to do another steps
You push undefined to result array `salaries[-1]`

Reply:
I got it. Thank you for explanation!

Reply:
@VladimirLQA please check the final solution cause it blocks merging
function getEmployeeInfo(name) {    
    if (!name || !name.length){ throw new Error("Invalid input")}
    const names = ["Bill", "Alex", "Anna", "Jhonn", "Antony"];
    const salaries = [1500, 3000, 1200, 1700, 850];
    const index = names.indexOf(name.trim());
    if (index === -1) return null;
    return [names[index], salaries[index]];
}